### ISSUE TITLE
Styling detail panel background-color when enableRowVirtualization is true 

### ISSUE THREAD

**Issue by jiiinCho** [KevinVandy/mantine-react-table#285]
### mantine-react-table version

v2.0.0-beta.0

### react & react-dom versions

v18.2.0

### Describe the bug and the steps to reproduce it

When using `renderDetailPanel` prop in combination with `enableRowVirtualization` set to true, there's a limitation where consumers cannot override the background color using the `mantineTableBodyRowProps`. This is because the background-color attribute is set in the `inner-virtual` style class, which is applied dynamically.

This issue can be solved with a workaround by using the class selector `mantine-Table-td-detail-panel` in the global scope. However, defining component styles in the global scope is generally not recommended due to maintenance and modularity concerns.

### Minimal, Reproducible Example - (Optional, but Recommended)

The code below references to https://github.com/KevinVandy/mantine-react-table/blob/v2/packages/mantine-react-table/stories/features/DetailPanel.stories.tsx#L11-L45
```diff
+ import classes from './DetailedPanel.module.css'; 

export const DetailPanelEnabled = () => (
  <MantineReactTable
    columns={[
      {
        accessorKey: 'firstName',
        header: 'First Name',
      },
      {
        accessorKey: 'lastName',
        header: 'Last Name',
      },
      {
        accessorKey: 'address',
        header: 'Address',
      },
    ]}
    data={[...Array(5)].map(() => ({
      address: faker.location.streetAddress(),
      city: faker.location.city(),
      firstName: faker.person.firstName(),
      lastName: faker.person.lastName(),
      phone: faker.phone.number(),
      state: faker.location.state(),
      zipCode: faker.location.zipCode(),
    }))}
+    enableRowVirtualization={true}
+    mantineTableBodyRowProps={({ isDetailPanel }) => {
+    return isDetailPanel
+     ? {
+          className: classes.detail,
+         }
+      : {};
+   }}
    renderDetailPanel={({ row }) => (
      <div style={{ display: 'grid' }}>
        <span>City: {row.original.city}</span>
        <span>State: {row.original.state}</span>
        <span>Zip: {row.original.zipCode}</span>
        <span>Phone: {row.original.phone}</span>
      </div>
    )}
  />
);
```
```css
/* DetailedPanel.module.css */
.detail {
  background-color: salmon;
}
```

### Screenshots or Videos (Optional)

storybook screenshot: 
<img width="1200" alt="Screenshot 2024-03-11 at 14 48 56" src="https://github.com/KevinVandy/mantine-react-table/assets/92979717/03e529e9-6503-4a5a-bcdc-5d1bc7972c52">

### Do you intend to try to help solve this bug with your own PR?

Yes, I think I know how to fix it and will discuss it in the comments of this issue

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by jiiinCho** [comment]
Since the detailed panel spans the entire row element, the `background-color` attribute can be applied to the `<tr />` element instead of individual `<td />` elements. For example:

```diff
/* packages/mantine-react-table/src/components/body/MRT_TableDetailPanel.module.css */
.root {
  display: table-row;
  width: 100%;
+ background-color: var(--mrt-base-background-color);
}

...
.inner-virtual {
- background-color: var(--mrt-base-background-color);
  transition: none;
}
```

But there is a chance that this modification can affect other behaviors since I'm quite new to the react-table library 😅. Please feel free to leave a comment!

**Comment by alessandrojcm** [comment]
Hey @jiiinCho feel free to open a PR with your fix and we can work it out from there :)

**Comment by jiiinCho** [comment]
Please have a look at [pull request](https://github.com/KevinVandy/mantine-react-table/pull/299) when you get a chance, thanks! 🎈

**Comment by alessandrojcm** [comment]
Solved in #299
